### ISSUE TITLE
Siamese network weird behaviour

### ISSUE THREAD

**Issue by swamiviv** [BVLC/caffe#2618]
This might be a silly question but here goes:

I have modified the mnist siamese network example to work for a set of image pairs that I have and I have trained that network. I am trying to use it in matlab.
Here is the strange thing, when I run the network and compute the loss I get different results for the same test pair. How is this possible once the network is trained ?
These are the lines I use to compute the loss:

`%Make pair data `
`pair_data = make_pair_data(img1,img2);`
`net.blobs('pair_data').set_data(pair_data);`
`net.forward_prefilled();`
`loss = net.blobs('loss').get_data()`

** Firstly, this produces different numbers each time I run.
** Secondly, if I reverse the pair, the loss value is different again. 

How are these possible ? Am I missing something simple here or is there a problem in the way the code was written ?

Any pointers would be great. Thanks!

**Comment by jshfeng** [comment]
This tracker is basically used for reporting bugs and issues with caffe. To ask questions about implementations, please refer to the mail list: https://groups.google.com/forum/#!forum/caffe-users Thanks!
